internal/storage/fs/snapshot.go

```diff
@@ -226,6 +227,7 @@ func documentsFromFile(fi fs.File, opts SnapshotOption) ([]*ext.Document, error)
 		if doc.Namespace == "" {
 			doc.Namespace = "default"
 		}
+		doc.Etag = fmt.Sprintf("%x-%x", stat.ModTime().Unix(), stat.Size())
```

Comment:
@erka im not 💯 that this is going to work for all declarative store types

for example it looks like for Git we arent setting the file modtime, so it seems like it would always return the default value?

https://github.com/flipt-io/flipt/blob/2bb2635fa56ff10536c9cc83649b4573ca4ad512/internal/gitfs/gitfs.go#L276-L277

i could be wrong though and we could be setting it elsewhere.. but perhaps for Git and OCI we should use the reference we have already?

we could potentially just build a reference for all declarative storage implementations, or use this method for object store and local FS implementations, and then the refs for Git and OCI

wdyt? cc @GeorgeMac

Reply:
@markphelps great catch. I have addressed the Git, OCI and Object stores.